pennylane/ops/qubit/matrix_ops.py

```diff
@@ -212,25 +212,22 @@ def __init__(
 
         self._padding_left = control_int * len(U)
         self._padding_right = 2 ** len(wires) - len(U) - self._padding_left
-        self._CU = None
 
+        # TODO[MARIA]: decide if it is too costly to compute this at creation
+        interface = qml.math.get_interface(self.U)
+        left_pad = qml.math.cast_like(qml.math.eye(self._padding_left, like=interface), 1j)
+        right_pad = qml.math.cast_like(qml.math.eye(self._padding_right, like=interface), 1j)
+        cu = qml.math.block_diag([left_pad, self.U, right_pad])
+
+        # make a QubitUnitary from the controlled matrix
+        new_params = list(params)
+        new_params[0] = cu
```

Comment:
This seems like something that should be in `compute_matrix`?

Reply:
whoops, this should be deleted!